README.md

```diff
@@ -1 +1,129 @@
-# angular-forms-readme
\ No newline at end of file
+# Built-in form validation methods
+
+## Overview
+
+You might've been thinking that Angular didn't have much more to offer - you were wrong! Angular helps us out massively with a quite common issue a lot of developers have to deal with - forms!
+
+## Objectives
+
+- Describe Angular form validation and states
+- Write a submit function
+- Create multiple inputs with different validation types
+- Display state Objects in the View
+
+## Angular Forms
+
+We're already two-way binding our input values to our controller via `ng-model`, but Angular allows us to take our form integration much, much further.
+
+When we assign a `name` to our form and input elements, Angular begins its magic and gives us access to a lot of useful tools.
+
+Let's take this exampe form:
+
+```html
+<form>
+	<input ng-model="ctrl.username" />
+	<input ng-model="ctrl.password" />
+</form>
+```
+
+And add name elements to our `<form>` and `<input />`s.
+
+```html
+<form name="form">
+	<input
+		name="username"
+		ng-model="ctrl.username" />
+	<input
+        name="password"
+        ng-model="ctrl.password" />
+</form>
+```
+
+We now get quite an awesome object added to our `$scope`:
+
+```js
+$scope.form = {
+	username: {
+		$untouched: true,
+		$touched: false,
+		$pristine: true,
+		$dirty: false,
+		$valid: true,
+		$invalid: false,
+		$error: {}
+	},
+	password: {
+        $untouched: true,
+        $touched: false,
+        $pristine: true,
+        $dirty: false,
+        $valid: true,
+        $invalid: false,
+        $error: {}
+    }
+};
+```
+
+As you can see we get quite a big, detailed object back. What do all of these mean?
+
+- `$untouched` - this is initially `true`, set to `false` when the user goes in and then out of the input
+- `$touched` - opposite of `$untouched`
+- `$pristine` - this is initially `true`, set to `false` when the user changes the input's value
```

Comment:
is `$pristine` different at all from `$untouched`? Same with `$touched` and `$dirty` - how are those different?